packages/form-js-viewer/src/render/components/form-fields/FilePicker.js

```diff
@@ -26,19 +26,28 @@ const type = 'filepicker';
 export function FilePicker(props) {
   /** @type {import("preact/hooks").Ref<HTMLInputElement>} */
   const fileInputRef = useRef(null);
-  /** @type {[File[],import("preact/hooks").StateUpdater<File[]>]} */
+  /** @type {[File[], import("preact/hooks").StateUpdater<File[]>]} */
   const [selectedFiles, setSelectedFiles] = useState([]);
+  /** @type {import('diagram-js/lib/core/EventBus').default} */
   const eventBus = useService('eventBus');
+  /** @type {import('../../FileRegistry').FileRegistry} */
+  const fileRegistry = useService('fileRegistry', false);
   const { field, onChange, domId, errors = [], disabled, readonly, required } = props;
   const { label, multiple = '', accept = '', id } = field;
   const evaluatedAccept = useSingleLineTemplateEvaluation(accept);
   const evaluatedMultiple =
     useSingleLineTemplateEvaluation(typeof multiple === 'string' ? multiple : multiple.toString()) === 'true';
   const errorMessageId = `${domId}-error-message`;
+  const filesKey = `${id}_value_key`;
```

Comment:
Keying needs to be done differently. It needs to include the full path otherwise dynamic list keys will overwrite eachother. The id is unique only at the design level. The true id of an element at the render level is the full path / key of the element.

![image](https://github.com/user-attachments/assets/fe0688cc-8107-47cc-a71f-58571c94d185)

My prefered fileKey would be something like `fileRef::${simpleHash(fullPath)}`. Hashing is optional but it's more about the fact that it's mental overhead for the users for it to be actually readable. Also I know some people will try to reverse engineer the path from it which is not something I'd like to encourage.

Reply:
I didn't do the hashing, I'm not sure if I agree with the mental overhead point.
Anyway, this can be easily fixed tomorrow.

Reply:
I guess on this one we can defer to @volodymyr-melnykc for a decision.